src/containers/GenerateStructureContainer.js

```diff
@@ -3,32 +3,56 @@ import { connect } from 'react-redux';
 import * as actions from '../actions/sm-data';
 import APIUtils from '../api/Utils';
 import * as showFormActions from '../actions/show-forms';
+import uuidv1 from 'uuid/v1';
 
 const apiUtils = new APIUtils();
 
 class GenerateStructureContainer extends Component {
-  constructor(props) {
-    super(props);
-    this.smData = [];
+  async componentDidMount() {
+    try {
+      const response = await apiUtils.getRequest('structure.json');
+      let structureJS = response.data;
+
+      // Add unique ids to every object
+      let smData = this.addIds([structureJS]);
+
+      // Update the redux store
+      this.props.buildSMUI(smData);
+    } catch (error) {
+      if (error.response !== undefined) {
+        this.props.handleResponse(error.response.status);
+      } else if (error.request !== undefined) {
+        this.props.handleResponse(error.request.status);
+      } else {
+        this.props.handleResponse(-1);
+      }
+    }
```

Comment:
@Dananji I re-factored this a little bit, to show a pattern of how we could implement `async / await`.  Ideally with this ES7 feature, we could do away with the `.then() { }` callbacks.